src/databricks/labs/lsql/dashboards.py

```diff
@@ -24,41 +25,59 @@
     Widget,
 )
 
-
-@runtime_checkable
-class _DataclassInstance(Protocol):
-    __dataclass_fields__: ClassVar[dict]
+T = TypeVar("T")
 
 
 class Dashboards:
     def __init__(self, ws: WorkspaceClient):
         self._ws = ws
 
-    def get_dashboard(self, dashboard_path: str):
+    def get_dashboard(self, dashboard_path: str) -> Dashboard:
         with self._ws.workspace.download(dashboard_path, format=ExportFormat.SOURCE) as f:
             raw = f.read().decode("utf-8")
             as_dict = json.loads(raw)
             return Dashboard.from_dict(as_dict)
 
-    def save_to_folder(self, dashboard_path: str, local_path: Path):
+    def save_to_folder(self, dashboard: Dashboard, local_path: Path):
         local_path.mkdir(parents=True, exist_ok=True)
-        dashboard = self.get_dashboard(dashboard_path)
-        better_names = {}
+
+        dashboard = self.with_better_names(dashboard)
+
         for dataset in dashboard.datasets:
-            name = dataset.display_name
-            better_names[dataset.name] = name
-            query_path = local_path / f"{name}.sql"
-            sql_query = dataset.query
-            self._format_sql_file(sql_query, query_path)
-        lvdash_yml = local_path / "lvdash.yml"
-        with lvdash_yml.open("w") as f:
-            first_page = dashboard.pages[0]
-            self._replace_names(first_page, better_names)
-            page = first_page.as_dict()
-            yaml.safe_dump(page, f)
-        assert True
-
-    def create_dashboard(self, dashboard_folder: Path) -> Dashboard:
+            query = self.format_query(dataset.query)
+            with (local_path / f"{dataset.name}.sql").open("w") as f:
+                f.write(query)
+
+        for page in dashboard.pages:
+            with (local_path / f"{page.name}.yml").open("w") as f:
+                yaml.safe_dump(page.as_dict(), f)
+
+    @staticmethod
+    def format_query(query: str) -> str:
+        try:
+            parsed_query = sqlglot.parse(query)
+        except sqlglot.ParseError:
+            formatted_query = query
+        else:
```

Comment:
This else i'm struggling to follow  else in try-except-else block. Can you move the else body to try?